### PR TITLE
fix(gatsby): restore client-only-paths behaviour

### LINKED ISSUE #15401: Gatsby@2.13.2 - Client-only routes broken
## Description

Client-only routes broken in Gatsby@2.13.2

### Steps to reproduce

Use starter project https://github.com/dwyfrequency/gatsby-ghub that is using the 'gatsby-plugin-create-client-paths' plugin and the latest version of Gatsby.

To check that version 2.13.2 is the problem, fix Gatsby version to 2.13.1 and check that it is now working as expected.

### Expected result

Client-only routes should work

### Actual result

Client-only routes not working

### Environment

System:
    OS: macOS 10.14.4
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.12.0 - ~/.nvm/versions/node/v11.12.0/bin/node
    Yarn: 1.15.2 - ~/.nvm/versions/node/v11.12.0/bin/yarn
    npm: 6.7.0 - ~/.nvm/versions/node/v11.12.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 75.0.3770.100
    Firefox: 67.0
    Safari: 12.1
  npmPackages:
    gatsby: ^2.10.0 => 2.13.3
    gatsby-image: ^2.0.20 => 2.2.4
    gatsby-plugin-create-client-paths: ^2.0.2 => 2.1.1
    gatsby-plugin-manifest: ^2.0.9 => 2.2.1
    gatsby-plugin-offline: ^2.0.16 => 2.2.0
    gatsby-plugin-playground: ^1.0.6 => 1.0.6
    gatsby-plugin-react-helmet: ^3.0.2 => 3.1.0
    gatsby-plugin-sharp: ^2.0.14 => 2.2.2
    gatsby-plugin-styled-components: ^3.1.0 => 3.1.0
    gatsby-source-filesystem: ^2.0.8 => 2.1.2
    gatsby-source-graphql: ^2.1.0 => 2.1.0
    gatsby-transformer-sharp: ^2.1.8 => 2.2.1
  npmGlobalPackages:
    gatsby: 2.7.3

### PR DESCRIPTION
**Pull request by pieh**
This PR is different attempt than https://github.com/gatsbyjs/gatsby/pull/15408 to fix regression introduced in https://github.com/gatsbyjs/gatsby/pull/13197

Here I expanded e2e tests to cover nested routers in user code. My previous PR would fail those. I also run them against 2.13.1 (last release before publish with breaking changes).

This is definitely messy and hacky (and I hate it), but I just don't see another way of preserving behaviour of Gatsby v2 and still having focus managment fixes in.

For context on how client-only-paths are broken right now please check https://github.com/gatsbyjs/gatsby/pull/15408#issuecomment-508600347 that goes over sample scenario and how this was handled before 2.13.2 (this PR restores that) and in 2.13.2

Fixes #15401

### CODE REVIEW

File: packages/gatsby/src/utils/webpack.config.js [hunk 1]
```diff
@@ -259,6 +259,18 @@ module.exports = async (program, directory, suppliedStage) => {
       rules.images(),
       rules.media(),
       rules.miscAssets(),
+
+      // This is a hack that exports one of @reach/router internals (BaseContext)
+      // to export list. We need it to reset basepath and baseuri context after
+      // Gatsby main router changes it, to keep v2 behaviour.
+      // We will need to most likely remove this for v3.
+      {
+        test: /reach\/router\/es\/index/,
```
Comment (wardpeet) [inline packages/gatsby/src/utils/webpack.config.js]: This supports windows & linux. If we want to test one file we can do it with require.resolve.
```suggestion
        test: require.resolve(`@reach/router/es/index`),
```

Comment (pieh) [reply packages/gatsby/src/utils/webpack.config.js]: Oh yeah, that makes much more sense. Thanks!

Comment (xavivars) [reply packages/gatsby/src/utils/webpack.config.js]: Quick question: why is this requiring `@reach/router/es/index` instead of `@reach/router/index`?

**Comment by jakubzloczewski** [comment]
@pieh, I think Link should be imported from '@reach/router'. Please correct me if I'm wrong or missing something.

https://github.com/gatsbyjs/gatsby/blob/561d33e2e491d3971cb2a404eec9705a5a493602/e2e-tests/development-runtime/src/pages/client-only-paths.js#L3

**Pull request merged**